src/Microsoft.DotNet.Darc/src/DarcLib/Helpers/LocalHelpers.cs

```diff
@@ -135,8 +136,12 @@ public static string GetRepoPathFromFolder(string gitLocation, string sourceFold
 
         ExecuteGitCommand(gitLocation, $"init {repoFolderName}", logger, workingDirectory);
 
+        string configKey = $"http.extraheader";
```

Comment:
I think it would be safer actually if we checkout and push in the same method. Because both checkout and push requires setting this environment variable.